tests/markets/polymarket/test_polymarket_data_models.py

```
@@ -0,0 +1,53 @@
+    return PolymarketGammaMarket(**defaults)  # type: ignore[arg-type]
+
+
+def test_token_ids_valid_json() -> None:
+    market = _make_gamma_market(clobTokenIds="[111, 222]")
+    assert market.token_ids == [111, 222]
+
+
+def test_token_ids_none_raises_value_error() -> None:
+    market = _make_gamma_market(clobTokenIds=None)
+    with pytest.raises(ValueError, match="Market has no token_ids"):
+        market.token_ids
```

Ruff (0.15.7)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)